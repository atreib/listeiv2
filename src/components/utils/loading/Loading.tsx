import React, { ReactChild, ReactChildren } from 'react';
import styled from 'styled-components';

interface ComponentProps {
  width?: string;
  height?: string;
  backgroundColor?: string;
  children: ReactChild | ReactChildren;
}

interface LoadingWrapperProps {
  width?: string;
  height?: string;
  backgroundColor?: string;
}

/*
	LoadingWrapper component

	Goal: Creates a fully-centralized with darker background loading panel
	Params (ComponentProps interface):
		- width?: (optional) width of our loading panel
		- height?: (optional) height of our loading panel
		- children: our loading panel content is a children element - react element
*/

const LoadingWrapper = styled.div<LoadingWrapperProps>`
  width: ${(props) => (props.width ? props.width : '100%')};
  height: ${(props) => (props.height ? props.height : '100%')};
  display: flex;
  align-items: center;
  justify-content: center;
  overflow: hidden;
  background: ${(props) => (props.backgroundColor ? props.backgroundColor : '#ccc')};
`;
LoadingWrapper.displayName = 'LoadingWrapper';

export const Loading = ({ children, width, height, backgroundColor }: ComponentProps) => {
  return (
    <LoadingWrapper width={width} height={height} backgroundColor={backgroundColor}>
      {children}
    </LoadingWrapper>
  );
};
