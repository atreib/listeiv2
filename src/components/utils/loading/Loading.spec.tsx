import React from 'react';
import { shallow } from 'enzyme';
import { Loading } from './Loading';

describe('Loading component test suite', () => {
  let mockChildren: JSX.Element;

  beforeEach(() => {
    mockChildren = <b>Mock</b>;
  });

  it('Should render correctly', () => {
    const wrapper = shallow(<Loading>{mockChildren}</Loading>);
    expect(wrapper).toMatchSnapshot();
  });
});
