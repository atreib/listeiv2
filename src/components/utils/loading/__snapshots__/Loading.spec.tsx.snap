// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component test suite Should render correctly 1`] = `
<LoadingWrapper>
  <b>
    Mock
  </b>
</LoadingWrapper>
`;
